services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/InMemoryKeyValueStorage.java

```diff
@@ -49,71 +52,65 @@ public void clear() {
   }
 
   @Override
-  public void close() {}
-
-  @Override
-  public boolean containsKey(final BytesValue key) throws StorageException {
+  public boolean containsKey(final byte[] key) throws StorageException {
     final Lock lock = rwLock.readLock();
     lock.lock();
     try {
-      return hashValueStore.containsKey(key);
+      return hashValueStore.containsKey(BytesValue.wrap(key));
     } finally {
       lock.unlock();
     }
   }
 
   @Override
-  public Optional<BytesValue> get(final BytesValue key) {
+  public Optional<byte[]> get(final byte[] key) throws StorageException {
     final Lock lock = rwLock.readLock();
     lock.lock();
     try {
-      return Optional.ofNullable(hashValueStore.get(key));
+      return Optional.ofNullable(hashValueStore.get(BytesValue.wrap(key)));
     } finally {
       lock.unlock();
     }
   }
 
   @Override
-  public long removeUnless(final Predicate<BytesValue> inUseCheck) {
-    final Lock lock = rwLock.writeLock();
-    lock.lock();
-    try {
-      long initialSize = hashValueStore.keySet().size();
-      hashValueStore.keySet().removeIf(key -> !inUseCheck.test(key));
-      return initialSize - hashValueStore.keySet().size();
-    } finally {
-      lock.unlock();
-    }
+  public long removeAllKeysUnless(final Predicate<byte[]> retainCondition) throws StorageException {
+    long initialSize = hashValueStore.keySet().size();
+    hashValueStore.keySet().removeIf(key -> !retainCondition.test(key.getArrayUnsafe()));
+    return initialSize - hashValueStore.keySet().size();
   }
 
   @Override
-  public Transaction startTransaction() {
-    return new InMemoryTransaction();
+  public void close() {}
```

Comment:
nitpick: `log.trace("close");` might help in some empty method warnings.

Reply:
That'd be inconsistent with this class and Pantheon more broadly, as predominantly we are not logging method entry or exists. (or resource closing)

Reply:
👍